subsys/bluetooth/audio/shell/bap_scan_delegator.c

```diff
@@ -18,60 +18,735 @@
+	if (state == NULL) {
+		shell_info(ctx_shell,
+			   "Could not get sync state from PA sync %p",
+			   sync);
+		return;
+	}
+
+	if (state->conn != NULL) {
+		bt_conn_unref(state->conn);
+		state->conn = NULL;
+	}
+
+	k_work_cancel_delayable(&state->pa_timer);
 }
 
-static void pa_recv(struct bt_bap_scan_delegator_recv_state *recv_state,
-		    const struct bt_le_per_adv_sync_recv_info *info,
-		    struct net_buf_simple *buf)
+static void pa_term_cb(struct bt_le_per_adv_sync *sync,
+		       const struct bt_le_per_adv_sync_term_info *info)
 {
-	char le_addr[BT_ADDR_LE_STR_LEN];
-	char hex[512];
+	struct sync_state *state;
 
-	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
-	bin2hex(buf->data, buf->len, hex, sizeof(hex));
-	shell_print(ctx_shell, "Receive state %p: device %s, tx_power %i, "
-		    "RSSI %i, CTE %u, data length %u, data %s",
-		    recv_state, le_addr, info->tx_power,
-		    info->rssi, info->cte_type, buf->len, hex);
+	shell_info(ctx_shell, "PA %p sync terminated", sync);
 
+	state = sync_state_get_by_pa(sync);
+	if (state == NULL) {
+		shell_error(ctx_shell,
+			    "Could not get sync state from PA sync %p",
+			    sync);
+		return;
+	}
+
+	if (state->conn != NULL) {
+		bt_conn_unref(state->conn);
+		state->conn = NULL;
+	}
+
+	k_work_cancel_delayable(&state->pa_timer);
 }
 
-static struct bt_bap_scan_delegator_cb cbs = {
-	.pa_synced = pa_synced,
-	.pa_term = pa_term,
-	.pa_recv = pa_recv
+static struct bt_le_per_adv_sync_cb pa_sync_cb =  {
+	.synced = pa_synced_cb,
+	.term = pa_term_cb,
 };
 
 static int cmd_bap_scan_delegator_init(const struct shell *sh, size_t argc,
 				       char **argv)
 {
-	bt_bap_scan_delegator_register_cb(&cbs);
+	bt_le_per_adv_sync_cb_register(&pa_sync_cb);
+	bt_bap_scan_delegator_register_cb(&scan_delegator_cb);
+	return 0;
+}
+static int cmd_bap_scan_delegator_set_past_pref(const struct shell *sh,
+						size_t argc, char **argv)
+{
+	bool past_pref;
+	int err;
+
+	err = 0;
+
+	past_pref = shell_strtobool(argv[1], 10, &err);
+	if (err != 0) {
+		shell_error(sh, "Failed to parse past_pref from %s", argv[1]);
+		return -ENOEXEC;
+	}
+
+	past_preference = past_pref;
+
+	return 0;
+}
+
+static int cmd_bap_scan_delegator_sync_pa(const struct shell *sh, size_t argc,
+					  char **argv)
+{
+	struct sync_state *state;
+	unsigned long src_id;
+	int err;
+
+	err = 0;
+
+	src_id = shell_strtoul(argv[1], 16, &err);
+	if (err != 0) {
+		shell_error(sh, "Failed to parse src_id from %s", argv[1]);
+
+		return -ENOEXEC;
+	}
+
+	if (src_id > UINT8_MAX) {
+		shell_error(sh, "src_id shall be 0x00-0xff");
+
+		return -ENOEXEC;
+	}
+
+	state = sync_state_get_by_src_id((uint8_t)src_id);
+	if (state == NULL) {
+		shell_error(ctx_shell, "Could not get state");
+
+		return -ENOEXEC;
+	}
+
+	if (past_preference &&
+	    state->past_avail &&
+	    state->conn != NULL) {
+		shell_info(sh, "Syncing with PAST");
+		err = pa_sync_past(state->conn, state, state->pa_interval);
+
+		if (err != 0) {
```

Comment:
```suggestion
		shell_info(sh, "Syncing with PAST");
		
		err = pa_sync_past(state->conn, state, state->pa_interval);
		if (err != 0) {
```
Wouldn't this be better structure?